src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -314,6 +347,47 @@ static OnHitEffect enemyPotionEffect(Supplier<MobEffectInstance> effect)
 		return (stack, target, attacker) -> target.addEffect(effect.get());
 	}
 	
+	/**
+	 * A function that causes a sweep effect (code based off of {@link #SWEEP}) and applies an array of effects on the target.
+	 * @param effects A varargs value, essentially an optional array of hit effects to be applied.
+	 */
+	static OnHitEffect sweepMultiEffect(OnHitEffect... effects)
+	{
+		return (stack, target, attacker) -> {
+			if(!(attacker instanceof Player playerAttacker))
+				return;
+			
+			boolean slowMoving = (double) (playerAttacker.walkDist - playerAttacker.walkDistO) < (double) playerAttacker.getSpeed();
+			boolean lastHitWasCrit = ServerEventHandler.wasLastHitCrit(playerAttacker);
+			
+			if(!(slowMoving
+					&& !lastHitWasCrit
+					&& playerAttacker.isOnGround()
+					&& playerAttacker.getAttackStrengthScale(0) >= 1))
+				return;
+			
+			float attackDamage = (float) playerAttacker.getAttribute(Attributes.ATTACK_DAMAGE).getValue();
+			float sweepEnchantMod = 1.0F + EnchantmentHelper.getSweepingDamageRatio(playerAttacker) * attackDamage;
+			
+			for(LivingEntity livingEntity : playerAttacker.level.getEntitiesOfClass(
+					LivingEntity.class, target.getBoundingBox().inflate(2.0D, 0.25D, 2.0D)))
+			{
+				if(!(livingEntity != playerAttacker
+						&& !playerAttacker.isAlliedTo(livingEntity)
+						&& (!(livingEntity instanceof ArmorStand) || !((ArmorStand) livingEntity).isMarker())
+						&& playerAttacker.distanceToSqr(livingEntity) < 9.0D))
```

Comment:
Double-inversions might make this harder to read, so I'd like to suggest this instead:
```suggestion
				if(livingEntity == playerAttacker
						|| playerAttacker.isAlliedTo(livingEntity)
						|| livingEntity instanceof ArmorStand stand && stand.isMarker()
						|| playerAttacker.distanceToSqr(livingEntity) >= 9.0D)
```

Reply:
Applied changes to both if statements that had this structure.